chirp/drivers/baofeng_1900.py

```diff
@@ -0,0 +1,596 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Andrew Morgan <ziltro@ziltro.com>
+# Copyright 2023 Edgar Tinajero <24572406+cetinajero@users.noreply.github.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import time
+import struct
+import unittest
+import logging
+
+from chirp import chirp_common, directory, memmap
+from chirp import bitwise, errors, util
+from chirp.settings import RadioSetting, RadioSettingGroup, \
+    RadioSettingValueInteger, RadioSettingValueList, \
+    RadioSettingValueBoolean, RadioSettings
+
+LOG = logging.getLogger(__name__)
+
+MEM_FORMAT = """
+#seekto 0x0010;
+struct {
+    lbcd rxfreq[4];
+    lbcd txfreq[4];
+    lbcd rxtone[2];
+    lbcd txtone[2];
+    u8 unknown3:1,
+       unknown2:1,
+       unknown1:1,
+       skip:1,
+       highpower:1,
+       narrow:1,
+       beatshift:1,
+       bcl:1;
+    u8 unknown4[3];
+} memory[16];
+#seekto 0x02B0;
+struct {
+    u8 voiceprompt;
+    u8 voicelanguage;
+    u8 scan;
+    u8 vox;
+    u8 voxlevel;
+    u8 voxinhibitonrx;
+    u8 lowvolinhibittx;
+    u8 highvolinhibittx;
+    u8 alarm;
+    u8 fmradio;
+} settings;
+#seekto 0x03C0;
+struct {
+    u8 unused:6,
+       batterysaver:1,
+       beep:1;
+    u8 squelchlevel;
+    u8 scanmode;
+    u8 timeouttimer;
+    u8 unused2[4];
+} settings2;
```

Comment:
Break this out into its own string:
```
H777_SETTINGS2 = """
 struct {
   ...
 } settings2;
```
and add one for yours:
```
BAOFENG1900_SETTINGS2 = """
struct {
 ...
} settings2;
```
Then see below in `process_mmap()`